src/components/YourCompany/OfficeAddresses.vue

```diff
@@ -620,15 +620,16 @@ export default class OfficeAddresses extends Mixins(CommonMixin) {
   }
 
   /**
-   * When stored office addresses change (ie, when data is loaded/updated/reset),
+   * When the component is mounted or stored office addresses change (ie, when data is loaded/updated/reset),
    * sets local properties and emits state events.
```

Comment:
Can you also explain why this component emits this event twice on init?

Reply:
Doesn't this already explain that? 
How would you word it?

Reply:
It only emits once on init (mounted due to immediate) and it emits again once the data changes (in our case, when the filing is loaded, and any subsequent changes). 

hehe i can write that.. though it seems long winded. I need your 'comment' expertise here!

Reply:
I'm not looking for "what" it's doing (ie, immediate + watching) but "why".

> It fires up front because the store values change with regards to the office addresses, every time a filing is loaded. Why it works initially but doesn't when it 'mounts again'. Because there is no changes to the store value at that point.

This sort of explained it for me but I will soon forget it, so hoping to get this into a code comment.

I guess the gist is that this component loads/inits/mounts before the data is loaded, and also after. Or something about the reactivity not set up to work as intended. I dunno.

Reply:
I'll try my best to do so without getting too wordy :)

Reply:
> It only emits once on init (mounted due to immediate) and it emits again once the data changes (in our case, when the filing is loaded, and any subsequent changes).

Yeah, this.

But now I'm wondering why this component is displayed before it had data 😛  (IMHO it shouldn't.) But that can be reworked another day.

Reply:
It's an internal issue in the component i think.
The component computes the values which it then feeds into the Base Address Sub component. This was only being called when the Store values changed. 
I think if the Base Addres sub components were utilizing the store getter directly it wouldn't have been an issue, but the Parent component is doing a local computation of sorts.